File: 632. 二叉树的最大节点.cpp
/**
 * Definition of TreeNode:
 * class TreeNode {
 * public:
 *     int val;
 *     TreeNode *left, *right;
 *     TreeNode(int val) {
 *         this->val = val;
 *         this->left = this->right = NULL;
 *     }
 * }
 */

class Solution {
public:
    /*
     * @param root: the root of tree
     * @return: the max node
     */
    void inorderTraversal(TreeNode* root) {
        if (root == nullptr) return;
        inorderTraversal(root->left);
        if (root->val > ret->val) ret = root;
        inorderTraversal(root->right);
    }

    TreeNode * maxNode(TreeNode * root) {
        ret = root;
        inorderTraversal(root);
        return ret;
    }

private:
    TreeNode* ret = nullptr;

};

// 总耗时 167 ms
// 您的提交打败了 78.80% 的提交!

File: Python3/632. 二叉树的最大节点.py
"""
Definition of TreeNode:
class TreeNode:
    def __init__(self, val):
        self.val = val
        self.left, self.right = None, None
"""

class Solution:
    """
    @param: root: the root of tree
    @return: the max node
    """
    def __init__(self) :
        self.max_node = None

    def findMax(self, root) :
        if root == None : return
        if root.val > self.max_node.val :
            self.max_node = root
        self.findMax(root.left)
        self.findMax(root.right)

    def maxNode(self, root):
        self.max_node = root
        self.findMax(root)
        return self.max_node

# 总耗时 1436 ms
# 您的提交打败了 22.96% 的提交!